src/Swarm/Game/State.hs

```diff
@@ -195,6 +196,7 @@ data GameState = GameState
   , _recipesIn :: IntMap [Recipe Entity]
   , _scenarios :: ScenarioCollection
   , _world :: W.World Int Entity
+  , _worldCursor :: Maybe W.Coords
```

Comment:
I guess this should go in the UIState, but the TUI.Model module does not import World, so maybe it's ok to keep this here?

Reply:
I don't see any problem with adding a `World` import to `TUI.Model`.